### PR TITLE
Fix historical vectors

### LINKED ISSUE #3299: Only plot one historical vectors across all realizations if they are equal
Pyqt plotter is currently adding the historical vector from the refcase and plot both the summaryvector and - if any - a historical vector. The summaryvector should be removed, ref https://github.com/equinor/ert/issues/3298 - and the historical vector should be added to all realizations, ref https://github.com/equinor/ert/issues/3297. The historical vectors are usually equal, but could be different in rare cases. Removing duplicates should (?) give better performance in the plotting.

### PR DESCRIPTION
**Pull request by frode-aarstad**
**Issue**
Resolves #3299 

Fix the displaying of historical data in the QT plot. 
Merge columns with equal history data

### CODE REVIEW

File: ert_gui/tools/plot/plot_api.py [hunk 1]
```diff
@@ -208,20 +209,31 @@ def _add_index_range(self, data):
...
-        index is the key."""
-        return self._facade.refcase_data(key)
-
-    def history_data(self, key, case=None):
+    def history_data(self, key, case=None) -> pd.DataFrame:
         """Returns a pandas DataFrame with the data points for the history for a
         given data key, if any.  The row index is the index/date and the column
         index is the key."""
 
+        if str(key).endswith("H") or "H:" in str(key):
```
Comment (oysteoh) [inline ert_gui/tools/plot/plot_api.py:217]: Should there be an explanation on this  piece of code? From my knowledge i think i understand the reason you added it - that being the cases where we actually have keys ending with `H`  - it is not an historical vector? The "loose" concept here is quite fragile, and i support adding this check...

Comment (frode-aarstad) [reply ert_gui/tools/plot/plot_api.py:217]: Yes, if you select an historical vector this should be picked up here. I can add a comment for this line

File: ert_gui/tools/plot/plot_api.py [hunk 2]
```diff
@@ -224,4 +219,18 @@ def history_data(self, key, case=None):
...
+
+        df = self.data_for_key(case, history_key)
+
+        if not df.empty:
+            df = df.T
+            # Drop columns with equal data
+            duplicate_cols = [
+                cc[1] for cc in combi(df.columns, r=2) if (df[cc[0]] == df[cc[1]]).all()
+            ]
+            duplicate_cols = [
```
Comment (oysteoh) [inline ert_gui/tools/plot/plot_api.py:231]: isn't this a reassignment of the 228 duplicate_cols..? I struggle a bit to understand how this should work :thinking:

Comment (frode-aarstad) [reply ert_gui/tools/plot/plot_api.py]: fixed

**Review by oysteoh — APPROVED** [review]

**Pull request merged**